optimizer-hints.md

```diff
@@ -201,6 +201,72 @@ EXPLAIN SELECT * FROM t WHERE EXISTS (SELECT /*+ SEMI_JOIN_REWRITE() */ 1 FROM t
 
 From the preceding example, you can see that when using the `SEMI_JOIN_REWRITE()` hint, TiDB can select the execution method of IndexJoin based on the driving table `t1`.
 
+### NO_DECORRELATE()
+
+The `NO_DECORRELATE()` hint tells the optimizer not to try to perform decorrelation for the correlated subquery in the specified query block. This hint is applicable to the `EXISTS`, `IN`, `ANY`, `ALL`, and `SOME` clauses that contain correlated columns, and scalar subqueries (that is, correlated subqueries).
```

Comment:
MySQL defines the correlated subquery as follows:
> A correlated subquery is a subquery that contains a reference to a table that also appears in the outer query.

The main point of "correlated subqueries" is "containing correlated columns". Currently, it could be a subquery started with `EXISTS`, `IN`, `ANY`, `ALL`, or `SOME`, or be a scalar subquery, so I added this description to the Chinese doc to make it more clear. And I think they are not "clauses" strictly.

```suggestion
The `NO_DECORRELATE()` hint tells the optimizer not to try to perform decorrelation for the correlated subquery in the specified query block. This hint is applicable to the `EXISTS`, `IN`, `ANY`, `ALL`, `SOME` subqueries and scalar subqueries that contain correlated columns (that is, correlated subqueries).
```